### ISSUE TITLE
Follow functionality independent of Friendship

### ISSUE THREAD

**Issue by luke-** [humhub/humhub#5089]
### Is your feature request related to a problem? Please describe.

Friendship and following should be separated from each other.

So far, a friendship behaves like a following and the actual follow functionality is no longer available between Friends.

### Describe the solution you'd like

Friendships should still trigger an automatic follow (migration!), but the Unfollow/Follow button should still be available.

---

This should also simplify stream queries, as friendships no longer need to be considered separately.

In the future, it might be possible to proceed in a similar way at Spaces level.

**Comment by yurabakhtin** [comment]
@luke- PR https://github.com/humhub/humhub/pull/5093.

**Comment by yurabakhtin** [comment]
still need to fix tests https://github.com/humhub/humhub/runs/2718388345?check_suite_focus=true

**Comment by yurabakhtin** [comment]
Fixed tests in commit https://github.com/humhub/humhub/pull/5093/commits/eb4e8b7dfb64de87f5ee7ba49814d19df279fc57

**Comment by luke-** [comment]
@yurabakhtin The relevant queries must also be removed from the dashboard stream query, right?

https://github.com/humhub/humhub/blob/enh/5089-independent-follow-of-friendship/protected/humhub/modules/dashboard/stream/filters/DashboardMemberStreamFilter.php#L82

**Comment by yurabakhtin** [comment]
@luke- I have reviewed the code and played with 4 different cases when user is Followed/Unfollowed + Friend/Not friend.

How it works currently with example:

- "User 1" has a Public post and a Private post on profile stream,
- "User 2" is Friend + Follow => Result:(what "User 2" sees on Dashboard stream)
  - Public + Private
- "User 2" is Friend + Unfollow =>
  - Public + Private
- "User 2" is Unfriend + Follow =>
  - only Public
- "User 2" is Unfriend + Unfollow =>
  - no posts

Please explain what result you expect to see on Dashboard instead of current results for each case, thanks.

**Comment by luke-** [comment]
@yurabakhtin If I am friends with a user but do not follow him. I should not see any posts from him on my dashboard.

**Comment by yurabakhtin** [comment]
@luke ok then I will implement this:

- Friend + Follow => Public + Private
- Friend + Unfollow => no posts
- Unfriend + Follow => only Public
- Unfriend + Unfollow => no posts

**Comment by luke-** [comment]
@yurabakhtin The screenshot looking good for me. FYI I've added another commit, now your last commit should also appear...

**Comment by yurabakhtin** [comment]
@luke- Yes, now I see my commit, thanks, but please see screenshot, probably all tests were executed together with next your commit:

<img width="895" alt="tests_not_passed" src="https://user-images.githubusercontent.com/6995524/120476095-dcbe8300-c3b2-11eb-8c6e-345acf8ea099.png">

**Comment by luke-** [comment]
@yurabakhtin this should be ok.

**Comment by miqrogroove** [comment]
After upgrading a site with friendship enabled, everyone is effectively unsubscribed from the Dashboard stream now, because nobody has followed anybody else.  This is a weird experience.  Do you need a new issue for that?

**Comment by luke-** [comment]
@miqrogroove Please open an issue for that.
@yurabakhtin Did we create a new follow record for friends?
